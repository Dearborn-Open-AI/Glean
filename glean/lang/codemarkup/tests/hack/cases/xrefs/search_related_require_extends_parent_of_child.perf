{
  "@generated": null,
  "hack.DeclarationSpan.6": 6,
  "hack.InterfaceDefinition.6": 6,
  "hack.TraitDefinition.6": 4
}