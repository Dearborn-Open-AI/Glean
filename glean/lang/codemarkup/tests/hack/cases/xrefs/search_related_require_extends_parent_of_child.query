query: codemarkup.SearchRelatedEntities { query = RequireExtendsParentOfChild, style = ShowAll }
perf: true
transform: [gensort, []]
