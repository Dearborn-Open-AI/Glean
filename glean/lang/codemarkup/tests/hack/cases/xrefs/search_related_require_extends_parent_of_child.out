[
  "@generated",
  {
    "key": {
      "query": 4,
      "parent": {
        "parent": {
          "hack": {
            "decl": {
              "container": {
                "class_": {
                  "key": { "name": { "key": { "name": { "key": "Machine" } } } }
                }
              }
            }
          }
        },
        "location": {
          "name": "Machine",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 264, "length": 135 } }
        }
      },
      "child": {
        "child": {
          "hack": {
            "decl": {
              "container": {
                "interface_": {
                  "key": { "name": { "key": { "name": { "key": "Jet" } } } }
                }
              }
            }
          }
        },
        "location": {
          "name": "Jet",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 893, "length": 75 } }
        }
      },
      "style": 0
    }
  },
  {
    "key": {
      "query": 4,
      "parent": {
        "parent": {
          "hack": {
            "decl": {
              "container": {
                "class_": {
                  "key": { "name": { "key": { "name": { "key": "Machine" } } } }
                }
              }
            }
          }
        },
        "location": {
          "name": "Machine",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 264, "length": 135 } }
        }
      },
      "child": {
        "child": {
          "hack": {
            "decl": {
              "container": {
                "trait": {
                  "key": { "name": { "key": { "name": { "key": "Plane" } } } }
                }
              }
            }
          }
        },
        "location": {
          "name": "Plane",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 519, "length": 242 } }
        }
      },
      "style": 0
    }
  },
  {
    "key": {
      "query": 4,
      "parent": {
        "parent": {
          "hack": {
            "decl": {
              "container": {
                "class_": {
                  "key": {
                    "name": { "key": { "name": { "key": "SuperEngine" } } }
                  }
                }
              }
            }
          }
        },
        "location": {
          "name": "SuperEngine",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 401, "length": 21 } }
        }
      },
      "child": {
        "child": {
          "hack": {
            "decl": {
              "container": {
                "interface_": {
                  "key": { "name": { "key": { "name": { "key": "Jet" } } } }
                }
              }
            }
          }
        },
        "location": {
          "name": "Jet",
          "file": { "key": "www/Require.php" },
          "location": { "span": { "start": 893, "length": 75 } }
        }
      },
      "style": 0
    }
  }
]